features/step_definitions/search_steps.rb

```diff
@@ -29,3 +29,41 @@
     page.should have_css('dt a span.match', text: text)
   end
 end
+
+Then /^they can filter the table by (?:name|system name)$/ do
+  @items = find_items
+  input = find('input[aria-label="Search input"]')
+  button = find('button[aria-label="Search"]')
+
+  input.set('ab')
+  button.click
+  assert_selector('.pf-c-popover__body', text: "To search, type at least 3 characters")
+
+  clear_search
+
+  input.set(@items.first)
+  button.click
+  assert_equal [@items.first], find_items
+
+  input.set(@items.last)
+  button.click
+  assert_includes find_items, @items.last
+
+  input.set('foooo')
+  button.click
+  assert_equal [], find_items
+
+  clear_search
+end
+
+def find_items
+  return all('td[data-label="System name"]').map(&:text).sort unless has_css?('th[data-label="Name"]')
+
+  all('td[data-label="Name"]').map(&:text).sort
+end
+
+def clear_search
+  clear_button = find('button[aria-label="Reset"]')
+  clear_button.click
+  assert_equal @items, find_items
+end
```

Comment:
The method is called _clear search_ but it also asserts the items, breaking SRP. I suggest to slip it into 2:
```
clear_search
assert_equal items, find_items
```